### PR TITLE
Tests for CASE/WHEN query with NULL support

### PR DESCRIPTION
**Pull request by abhioncbr**
This PR is for the [issue](https://github.com/apache/pinot/issues/10513) 
- enables `ENABLE_NULL_HANDLING` for multistage query testing.
- Added multiple case/when queries with null literal function case.

cc: @shenyu0127, @Jackie-Jiang

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/CaseTransformFunction.java [hunk 1]
```diff
@@ -331,7 +331,7 @@ protected int[] transformToIntValuesSVUsingValue(ValueBlock valueBlock) {
       }
     }
     if (!unselectedDocs.isEmpty()) {
-      if (_elseStatement == null) {
+      if (_elseStatement == null || LiteralTransformFunction.isNullLiteralTransform(_elseStatement)) {
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/CaseTransformFunction.java:334]: To simplify this, we can change `constructStatementListCalcite()` (also change `constructStatementListLegacy()` for backward compatible) to set `_elseStatement` to `null` when it is null literal

File: pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/LiteralTransformFunction.java [hunk 2]
```diff
@@ -269,4 +277,15 @@ public RoaringBitmap getNullBitmap(ValueBlock valueBlock) {
     bitmap.add(0L, length);
     return bitmap;
   }
+
+  /*
+   * Util function to check whether the Literal transform is null or not.
+   */
+  public static boolean isNullLiteralTransform(TransformFunction function) {
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/LiteralTransformFunction.java:284]: Suggest adding  `boolean isNull()` instead. Checking `_dataType` itself should be enough (consistent with `getNullBitmap()`)

**Review by Jackie-Jiang — COMMENTED** [review]
@shenyu0127 Please also take a look

**Comment by shenyu0127** [comment]
Without this PR, the query `SELECT CASE WHEN boolCol = true THEN 'Yes'` also returns NULL if `boolCol` is NULL or false.  Is the purpose of this PR only to improve performance? If that's the case, we can simply re-enable the ignored query.

**Comment by abhioncbr** [comment]
> > > Without this PR, the query `SELECT CASE WHEN boolCol = true THEN 'Yes'` also returns NULL if `boolCol` is NULL or false. Is the purpose of this PR only to improve performance? If that's the case, we can simply re-enable the ignored query.
> > 
> > 
> > Initially, when I started looking a couple of months back, it was an issue, but as of now, it is more or less a performance improvement and enabling the ignored queries. Please let me know if we can drop the performance improvement changes. Thanks
> 
> I see. Please drop the performance improvement changes.

Just out of curiosity, any particular reason for not having the performance logic?

**Comment by shenyu0127** [comment]
> > > > Without this PR, the query `SELECT CASE WHEN boolCol = true THEN 'Yes'` also returns NULL if `boolCol` is NULL or false. Is the purpose of this PR only to improve performance? If that's the case, we can simply re-enable the ignored query.
> > > 
> > > 
> > > Initially, when I started looking a couple of months back, it was an issue, but as of now, it is more or less a performance improvement and enabling the ignored queries. Please let me know if we can drop the performance improvement changes. Thanks
> > 
> > 
> > I see. Please drop the performance improvement changes.
> 
> Just out of curiosity, any particular reason for not having the performance logic?

We prefer one PR to do one simple thing.

- If it's to fix the bug and re-enable the ignored query, the two changes are closely related and can be put one PR.
- If it's to improve performance and re-enable the ignored query, the two changes are not related and can be put in two PRs separately. Putting these two things in one PR causes confusion to the reviewer.

**Comment by abhioncbr** [comment]
As suggested, diving this PR into two
- for enabling the ignored query
- for performance changes. 
Thanks

**Review by Jackie-Jiang — APPROVED** [review]
LGTM

Could you revise the title and description for the PR accordingly?

**Review by shenyu0127 — APPROVED** [review]
LGTM

**Pull request merged**